### PR TITLE
feat: support customized variant name for fragment variants

### PR DESCRIPTION
**Pull request by FentPams**
## Description
Leverage"Variant" entry in manifest to support variant name customization for fragment experimentation. 

Untitled (1)
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by FentPams** [comment]
@ramboz I am trying to leverage the _variant_ column  in spreadsheet, extract it as a customized variant name. 

Before implementing that, could you help me confirm:
 1. Is Variant column designed to customize variant name? (I didn't see clearly documentation on this part, and not being used in code either)
 2. If the answer for Q1 is yes, then could we change the cell "`Variant`" into "`Names`", so we can align the implementation from page-level name customization. The pros for implementing in this way is: No code change needed, with only this one line which we already have```const labelNames = stringToArray(metadata.names);``` , while we do have cons: The naming is inconsistent (we take the variant column as '`label`' in the code level), which could cause confusion.
 3. And if the solution above is not good, we can change both, in page level, change the "`Experiment Names` => `Experiment Labels`" and in spreadsheet, from "`Variant` => `Labels`", which is consistent on code level and authoring level, while it requires code change and "label" sounds not as intuitive as "names" for authors.
 
I'd like to have your thoughts. Thank you :)

**Comment by FentPams** [comment]
Also do you have idea why lint fails here? I only made very little change 🤔

**Comment by ramboz** [comment]
1. The variant is rather used for the variant "id" so you can map each variant to a URL.
2. Since you used `Names` for the variant name in the metadata, I'd use the same for the column name and update the `depluralizeProps` call so it also covers `name` so we have both singular and plural automatically
3. Names is good

**Comment by FentPams** [comment]
@ramboz  Thank you for reviewing, I made the change based on your suggestion. Could you help review that?
And I noticed the test still fail, but where the err comes are not same as the one before you fixed.  I tried to change the textContent as its expected, but seems not the right way to fix..😅 Could you teach me how to fix that? Thank you!

**Comment by ramboz** [comment]
The tests are a bit "flaky" (i.e. we have some race conditions and sometimes the tests evaluate too soon), and I think you just hit a bad run here. Let me try re-running the tests

File: src/index.js [hunk 1]
```diff
@@ -659,9 +659,9 @@ async function getExperimentConfig(pluginOptions, metadata, overrides) {
  */
 function parseExperimentManifest(entries) {
   return Object.values(Object.groupBy(
-    entries.map((e) => depluralizeProps(e, ['experiment', 'variant', 'split'])),
+    entries.map((e) => depluralizeProps(e, ['experiment', 'name', 'split'])),
     ({ experiment }) => experiment,
-  )).map(aggregateEntries('experiment', ['split', 'url', 'variant']));
+  )).map(aggregateEntries('experiment', ['split', 'url', 'name']));
```
Comment (ramboz) [inline src/index.js:664]: @FentPams We actually need both the `variant` which defines the identifier, and `name` which provides the user friendly label in the UI

Comment (ramboz) [reply src/index.js:664]: Also you can see that the `Supports plural format for manifest keys` test is failing, which means that change introduced a regression.
https://github.com/adobe/aem-experimentation/blob/v2/tests/experiments.test.js#L238-L245

If you check https://github.com/adobe/aem-experimentation/blob/v2/tests/fixtures/experiments/fragments--alt.json#L6, you'll see we actually test for `Variants` explicitly.

We can update the 2 json files with the name and names options, so we validate the new feature as well following something like https://github.com/adobe/aem-experimentation/blob/v2/tests/experiments.test.js#L277-L293.

Ping me directly so I can get you started on the test framework :)

Comment (FentPams) [reply src/index.js]: @ramboz  Thank you, I finally got it work 🥹 Could you help review this?

**Review by ramboz — APPROVED** [review]

**Comment by FentPams** [comment]
The last commit is to fix the label text in overlay pill. The bug is introduced by changing the "`Variant`" into "`Names`" in manifest. In`decorateExperimentPill()` in [preview.js](https://github.com/adobe/aem-experimentation/blob/7efc8edbebaab220cd488465e8db82f6d953c7b1/src/preview.js#L348C7-L348C27), the label took the value from` label: metadata.name || Experiment ${metadata.value || metadata.experiment},`. In the case we have "Names" defined in manifest, the overlay pill will looks like this:
Example Image

After checking the three cases, page, section, fragment in experiments. I found the `metada.name` is not used. The section and page all using `metadata.value`, while fragment uses `metadata.experiment`:
Untitled (7)Untitled (6)

So, I am thinking of removing the `metadata.name`:
Untitled (6)

I am not sure whether this metada.name will be used, or will leads to side effect after removing it. I'd like to have your thoughts here. Thank you! @ramboz

**Comment by ramboz** [comment]
Good analysis! The `metadata.name` was used in the past for a different format of Excel manifests that we don't have in v2 anymore

**Pull request merged**